### PR TITLE
FUSETOOLS2-1670 - improve message on non verified breakpoints

### PR DESCRIPTION
**Pull request by apupier**
imptovedMessageOnBreakpoint

### CODE REVIEW

File: src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java [hunk 1]
```diff
@@ -169,6 +170,14 @@ private SetBreakpointsResponse setBreakpointsSync(SetBreakpointsArguments setBre
 		return response;
 	}
 
+	private void addBreakpoint(SourceBreakpoint sourceBreakpoint, String nodeId) {
+		if (sourceBreakpoint.getCondition() != null) {
+			connectionManager.getBacklogDebugger().addConditionalBreakpoint(nodeId, CAMEL_LANGUAGE_SIMPLE, sourceBreakpoint.getCondition());
```
Comment (joshiraez) [inline src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java:175]: Prefer having positive conditions to negative conditions.

It's also better to have

- Preconditions
- Happy path

As it follows natural reading path

```
if (precondition == null) {
   DoSomething
}

DoSometingIfEverythingIsOk

```

File: src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			
+			@Override
+			public void addLifecycleInterceptor(RouteBuilderLifecycleStrategy interceptor) {
+				super.addLifecycleInterceptor(interceptor); // XXX-breakpoint-XXX
+			}
+		});
+	}
+	
+	@Test
+	void testBreakpointSetOutisdeOfRoute() throws Exception {
```
Comment (joshiraez) [inline src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java:50]: Please, organize the test following the Given, Then, Path convention as it's hard to see what is what.

Is not required to put comments but at least separate the test in 3 parts separated by line breaks

File: src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+					.log("Log from test").id(logEndpointId);
+			}
+			
+			@Override
+			public void addLifecycleInterceptor(RouteBuilderLifecycleStrategy interceptor) {
+				super.addLifecycleInterceptor(interceptor); // XXX-breakpoint-XXX
+			}
+		});
+	}
+
```
Comment (joshiraez) [inline src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java:48]: Missing one test/two test for the three different elses in the code? Please add them

File: src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java [hunk 4]
```diff
@@ -148,18 +147,20 @@ private SetBreakpointsResponse setBreakpointsSync(SetBreakpointsArguments setBre
...
 					}
 				} catch (Exception e) {
-					LOGGER.warn("Cannot find related id for "+ source.getPath() + " l." + line, e);
+					String baseMessage = "The Camel debugger cannot find the related id for "+ source.getPath() + " l." + line;
+					breakpoint.setMessage(baseMessage + ". See logs for more details on exception that occured: "+ e.getMessage());
+					LOGGER.warn(baseMessage, e);
 				}
 			} else {
-				LOGGER.warn("No active routes find in Camel context. Consequently, cannot set breakpoint for {} l.{}", source.getPath(), line);
+				String message = "No active routes found in Camel context. Consequently, the Camel debugger cannot set breakpoint for "+source.getPath()+" l."+line;
```
Comment (joshiraez) [inline src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java:161]: The string literals should be in constants or functions.

As these have some values been used, is better to put a closure or function that returns this string

File: src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java [hunk 5]
```diff
@@ -148,18 +147,20 @@ private SetBreakpointsResponse setBreakpointsSync(SetBreakpointsArguments setBre
...
-							connectionManager.getBacklogDebugger().addBreakpoint(nodeId);
-						}
+						addBreakpoint(sourceBreakpoint, nodeId);
 						breakpoint.setVerified(true);
+					} else {
+						breakpoint.setMessage("The Camel debugger cannot find the related id for "+ source.getPath() + " l." + line);
 					}
 				} catch (Exception e) {
-					LOGGER.warn("Cannot find related id for "+ source.getPath() + " l." + line, e);
+					String baseMessage = "The Camel debugger cannot find the related id for "+ source.getPath() + " l." + line;
```
Comment (joshiraez) [inline src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java:156]: The string literals should be in constants or functions.

As these have some values been used, is better to put a closure or function that returns this string

This is also duplicated with line 153 so extra important

**Review by joshiraez — CHANGES_REQUESTED** [review]

File: src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java (hunk 2 above)
Comment (joshiraez) [reply src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java:50]: https://martinfowler.com/bliki/GivenWhenThen.html

Comment (apupier) [reply src/test/java/com/github/cameltooling/dap/internal/UnverifiedBreakpointTest.java:50]: there are already empty lines to separate the given, when, then

File: src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java (hunk 1 above)
Comment (apupier) [reply src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java:175]: usually happy path is when a value is not null. I think using `!= null` is very widespread in Java world.

In this case, it is more 2 different paths, than one happy and one handling error or missing information.

So it is:

if (case1) {
   DoSomethingInCase1
} else {
   DoSomethingInCase2
}

This is anyway not code modified in this PR, it is only extracted from existing code to reduce cyclomatic complexity on another function for which the threshold recommended by Sonar has been hit. Created https://issues.redhat.com/browse/FUSETOOLS2-1831then

File: src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java (hunk 5 above)
Comment (apupier) [reply src/main/java/com/github/cameltooling/dap/internal/CamelDebugAdapterServer.java]: Good point on the duplication. I modified the message to be more precise. it is slightly different cases and better to have more precise message

**Review by joshiraez — APPROVED** [review]

**Comment by joshiraez** [comment]
Thanks for adding the suggestions! It looks perfect now

**Pull request merged**